web/src/message_list_data.ts

```diff
@@ -300,7 +318,8 @@ export class MessageListData {
             // is a (1) sorted, and (2) consecutive block of
             // messages that belong in this message list; those
             // facts should be ensured by the caller.
-            if (this.empty() || msg.id > this.last_including_muted().id) {
+            const last_id = this.last_including_muted()?.id;
+            if (this.empty() || (last_id && msg.id > last_id)) {
```

Comment:
```ts
            const last = this.last_including_muted();
            if (last === undefined || msg.id > last.id) {
```

since `this.empty()` if and only if `last === undefined` (and we don’t want to confuse `undefined` with `0`).